src/Google.Bigquery.V2/BigqueryClient.cs

```diff
@@ -0,0 +1,164 @@
+        /// using this client's project ID.
+        /// </summary>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="tableId">The table ID. Must not be null.</param>
+        /// <returns>A <see cref="TableReference"/> representing the requested table.</returns>
+        public TableReference GetTableReference(string datasetId, string tableId) =>
+            GetTableReference(ProjectId, datasetId, tableId);
+
+        /// <summary>
+        /// Creates a <see cref="TableReference"/> from the given project ID, dataset ID and table ID.
+        /// </summary>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="tableId">The table ID. Must not be null.</param>
+        /// <returns>A <see cref="TableReference"/> representing the requested table.</returns>
+        public TableReference GetTableReference(string projectId, string datasetId, string tableId) =>
+            new TableReference
+            {
+                ProjectId = Preconditions.CheckNotNull(projectId, nameof(projectId)),
+                DatasetId = Preconditions.CheckNotNull(datasetId, nameof(datasetId)),
+                TableId = Preconditions.CheckNotNull(tableId, nameof(tableId)),
+            };
+
+        /// <summary>
+        /// Creates a <see cref="DatasetReference"/> from the given dataset ID,
+        /// using this client's project ID.
+        /// </summary>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <returns>A <see cref="DatasetReference"/> representing the requested dataset.</returns>
+        public DatasetReference GetDatasetReference(string datasetId) =>
+            GetDatasetReference(ProjectId, datasetId);
+
+        /// <summary>
+        /// Creates a <see cref="DatasetReference"/> from the given project ID and dataset ID.
+        /// </summary>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <returns>A <see cref="DatasetReference"/> representing the requested dataset.</returns>
+        public DatasetReference GetDatasetReference(string projectId, string datasetId) =>
+            new DatasetReference
+            {
+                ProjectId = Preconditions.CheckNotNull(projectId, nameof(projectId)),
+                DatasetId = Preconditions.CheckNotNull(datasetId, nameof(datasetId)),
+            };
+
+        /// <summary>
+        /// Creates a <see cref="JobReference"/> from the given job ID,
+        /// using this client's project ID.
+        /// </summary>
+        /// <param name="jobId">The job ID. Must not be null.</param>
+        /// <returns>A <see cref="JobReference"/> representing the requested job.</returns>
+        public JobReference GetJobReference(string jobId) =>
```

Comment:
Should there be an overload of this method that takes a projectId?
As in the GetTableReference and GetDatasetReference methods above.